src/main/scala/rocket/CSR.scala

```diff
@@ -626,18 +627,18 @@ class CSRFile(
   val debug_csrs = if (!usingDebug) LinkedHashMap() else LinkedHashMap[Int,Bits](
     CSRs.dcsr -> reg_dcsr.asUInt,
     CSRs.dpc -> readEPC(reg_dpc).sextTo(xLen),
-    CSRs.dscratch -> reg_dscratch.asUInt) ++
+    CSRs.dscratch0 -> reg_dscratch0.asUInt) ++
     reg_dscratch1.map(r => CSRs.dscratch1 -> r)
 
   val read_mnstatus = WireInit(0.U.asTypeOf(new MNStatus()))
   read_mnstatus.mpp := reg_mnstatus.mpp
   read_mnstatus.mpv := reg_mnstatus.mpv
   read_mnstatus.mie := reg_rnmie
   val nmi_csrs = if (!usingNMI) LinkedHashMap() else LinkedHashMap[Int,Bits](
-    CSRs.mnscratch -> reg_mnscratch,
-    CSRs.mnepc -> readEPC(reg_mnepc).sextTo(xLen),
-    CSRs.mncause -> reg_mncause,
-    CSRs.mnstatus -> read_mnstatus.asUInt)
+    CustomCSRs.mnscratch -> reg_mnscratch,
+    CustomCSRs.mnepc -> readEPC(reg_mnepc).sextTo(xLen),
+    CustomCSRs.mncause -> reg_mncause,
+    CustomCSRs.mnstatus -> read_mnstatus.asUInt)
```

Comment:
I wonder whether this should use the customCSR API:
https://github.com/chipsalliance/rocket-chip/blob/5ebac8f75926677f026cb8dab5181503300b28c6/src/main/scala/rocket/CSR.scala#L359

Reply:
Review in RC Workshop suggested not using customCSRs API, which is for "really" custom CSRs. The current custom instructions/CSRs in `Instructions.scala` is just not ratified.

In my opinion, using this API with NMI logics is not trivial.

Review also suggested submodule riscv-opcodes and auto generate `Instructions.scala`.

I think these can be separated into a file named `NotRatifiedInstructions.scala` to distinguish those existing API named CustomCSRs (e.g. in `CustomCSRs.scala`). When instructions get ratified, they can be removed from this file and riscv-opcodes submodule can bump to cover them.